packages/service-library/src/servicelib/file_utils.py

```diff
@@ -1,13 +1,37 @@
 import asyncio
+import shutil
 from pathlib import Path
 
+# https://docs.python.org/3/library/shutil.html#shutil.rmtree
+# https://docs.python.org/3/library/os.html#os.remove
+from aiofiles.os import remove
+from aiofiles.os import wrap as sync_to_async
 
-async def remove_directory(path: Path, only_children: bool = False) -> None:
+_shutil_rmtree = sync_to_async(shutil.rmtree)
+
+
+async def _rm(path: Path, ignore_errors: bool = False):
+    """Removes file or directory"""
+    try:
+        await remove(path)
+    except IsADirectoryError:
+        await _shutil_rmtree(path, ignore_errors=ignore_errors)
+
+
+async def remove_directory(
+    path: Path, only_children: bool = False, ignore_errors: bool = False
+) -> None:
     """Optional parameter allows to remove all children and keep directory"""
+    if not path.exists():
+        if ignore_errors:
+            return
+
+        raise FileNotFoundError(f"No such file or directory {path}")
```

Comment:
Yes, I've done some testing and will actually add a comment. To keep them in here. Since the exceptions do not get raised.

Reply:
this test shows differently. 

``shutil.rmtree`` and ``os.remove`` raising ``FileNotFoundError``

``` python
>>> import shutil, os
>>> from pathlib import Path
>>> p = Path("path/to/fakedir")
>>> shutil.rmtree(p, ignore_errors=True)
>>> shutil.rmtree(p)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/crespo/.pyenv/versions/3.8.10/lib/python3.8/shutil.py", line 709, in rmtree
    onerror(os.lstat, path, sys.exc_info())
  File "/home/crespo/.pyenv/versions/3.8.10/lib/python3.8/shutil.py", line 707, in rmtree
    orig_st = os.lstat(path)
FileNotFoundError: [Errno 2] No such file or directory: 'path/to/fakedir'
>>> os.remove(p)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
FileNotFoundError: [Errno 2] No such file or directory: 'path/to/fakedir'
>>>
```

``shutil.rmtree`` raising ``NotADirectoryError``

```python
>>> p.touch()
>>> shutil.rmtree(p)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/crespo/.pyenv/versions/3.8.10/lib/python3.8/shutil.py", line 718, in rmtree
    _rmtree_safe_fd(fd, path, onerror)
  File "/home/crespo/.pyenv/versions/3.8.10/lib/python3.8/shutil.py", line 631, in _rmtree_safe_fd
    onerror(os.scandir, path, sys.exc_info())
  File "/home/crespo/.pyenv/versions/3.8.10/lib/python3.8/shutil.py", line 627, in _rmtree_safe_fd
    with os.scandir(topfd) as scandir_it:
NotADirectoryError: [Errno 20] Not a directory: PosixPath('foo')

```

Reply:
Due to a purest issue I was unable to verify what you are saying within the test suite. But it actually works as expected

Since I do not want to spend any more time here, I disabled the 2 failing tests